### ISSUE TITLE
[BUG] Memory access error (memory dumped)

### ISSUE THREAD

**Issue by Linus789** [CCExtractor/ccextractor#1258]
CCExtractor version:
```
CCExtractor 0.88, Carlos Fernandez Sanz, Volker Quetschke.
Teletext portions taken from Petr Kutalek's telxcc
--------------------------------------------------------------------------
CCExtractor detailed version info
        Version: 0.88
        Git commit: 1534d81ae75c6562fb33dcd58050d90f3e54a4b7
        Compilation date: 2020-04-23
        File SHA256: bbbb5ed6264b84a4fb91e4657c569117fd884a2730d763764d8a75d05808f198
Libraries used by CCExtractor
        Tesseract Version: 4.0.0-beta.1
        Leptonica Version: leptonica-1.75.3
        libGPAC Version: 0.7.2-DEV
        zlib: 1.2.11
        utf8proc Version: 2.4.0
        protobuf-c Version: 1.3.1
        libpng Version: 1.6.35
        FreeType 
        libhash
        nuklear
        libzvbi
```

# In raising this issue, I confirm the following:

- [x] I have read and understood the [contributors guide](https://github.com/CCExtractor/ccextractor/blob/master/.github/CONTRIBUTING.md).
- [x] I have checked that the bug-fix I am reporting can be replicated, or that the feature I am suggesting isn't already present.
- [x] I have checked that the issue I'm posting isn't already reported.
- [x] I have checked that the issue I'm porting isn't already solved and no duplicates exist in [closed issues](https://github.com/CCExtractor/ccextractor/issues?q=is%3Aissue+is%3Aclosed) and in [opened issues](https://github.com/CCExtractor/ccextractor/issues)
- [x] I have checked the pull requests tab for existing solutions/implementations to my issue/suggestion.
- [x] I have used the latest available version of CCExtractor to verify this issue exists.
- [x] I have ticked all the boxes in this section and to prove it I'm deleting the section completely to remove boilerplate text.

# Necessary information

- Is this a regression (i.e. did it work before)? NO
- What platform did you use? Linux
- What were the used arguments? `./ccextractor -hardsubx video.mp4 -o ocred.ssa`

# Video links

* https://drive.google.com/open?id=1kJHlh1CtotjmLpmvJdzwmY7ciDjHjEZK
Basically not working with any video.

# Additional information

Built CCExtractor with CMake
`mkdir build && cd build; cmake -DWITH_OCR=on -DWITH_HARDSUBX=on ../src/ && make`

```
$ ./ccextractor -hardsubx /home/linus/Videos/out.mp4 -o s0e1.ass
CCExtractor 0.88, Carlos Fernandez Sanz, Volker Quetschke.
Teletext portions taken from Petr Kutalek's telxcc
--------------------------------------------------------------------------
HardsubX (Hard Subtitle Extractor) - Burned-in subtitle extraction subsystem
Speicherzugriffsfehler (Speicherabzug geschrieben)
```
`Memory access error (memory dumped)`

**Comment by cfsmp3** [comment]
Not a lot of info there. I'd try compiling it with debug info and running it with valgrind. Hopefully the stacktrace will be helpful.

Also, try writing to a different format - with a with of luck, .srt will work on the same input. Fixing output is always easier.

**Comment by Linus789** [comment]
[cmake_debug.txt](https://github.com/CCExtractor/ccextractor/files/4522823/cmake_debug.txt)
[make_debug.txt](https://github.com/CCExtractor/ccextractor/files/4522824/make_debug.txt)
[valgrind_debug.txt](https://github.com/CCExtractor/ccextractor/files/4522826/valgrind_debug.txt)

**Comment by muzena** [comment]
@cfsmp3 
I have the same problem on Linux Mint 19.
I have mkv PVR records, recorded in [Tvheadend](https://github.com/tvheadend/tvheadend), with [Happuage WinTV-HVR-2205](https://www.hauppauge.co.uk/site/products/data_hvr2205board.html) 
I tried to extract from mkv file subtitles it gives me Segmentation fault and Memory write error
I compile with these flags:
`     ./configure --enable-hardsubx --enable-ocr --enable-ffmpeg --with-gui`
I tried with ts file it gave me same error.

CCExtractor version: 
CCExtractor 0.88, Carlos Fernandez Sanz, Volker Quetschke.
Teletext portions taken from Petr Kutalek's telxcc
CCExtractor detailed version info
	Version: 0.88
	Git commit: 28dd35b040595435a8d15bf084e00b528ba2789e
	Compilation date: 2020-05-02
	File SHA256: 905b6e0ba15b38ea1067514688b9257c8462739d9b4cef34718fee93d4a9049b
Libraries used by CCExtractor
	Tesseract Version: 4.0.0-beta.1
	Leptonica Version: leptonica-1.75.3
	libGPAC Version: 0.7.2-DEV
	zlib: 1.2.11
	utf8proc Version: 2.4.0
	protobuf-c Version: 1.3.1
	libpng Version: 1.6.35
	FreeType 
	libhash
	nuklear
	libzvbi

[valgrind-v.txt](https://github.com/CCExtractor/ccextractor/files/4567859/valgrind-v.txt)
[MKV file](https://drive.google.com/open?id=1BS_TfE_u7NfxZN5ZT3q28gI98F4xhxfq)

**Comment by MackeyStingray** [comment]
Seeing the same issue in Debian Buster.  The issue is
```
/home/linus/Programme/CCExtractor/ccextractor/src/lib_ccx/hardsubx.c:228:18: warning: implicit declaration of function ‘probe_tessdata_location’ [-Wimplicit-function-declaration]
  tessdata_path = probe_tessdata_location(lang);
                  ^~~~~~~~~~~~~~~~~~~~~~~
/home/linus/Programme/CCExtractor/ccextractor/src/lib_ccx/hardsubx.c:228:16: warning: assignment makes pointer from integer without a cast [-Wint-conversion]
  tessdata_path = probe_tessdata_location(lang);
                ^
/home/linus/Programme/CCExtractor/ccextractor/src/lib_ccx/hardsubx.c:238:17: warning: assignment makes pointer from integer without a cast [-Wint-conversion]
   tessdata_path = probe_tessdata_location("eng");
```

Adding the line `char* probe_tessdata_location(const char *lang);` to src/lib_ccx/ocr.h resolves it.

**Comment by cfsmp3** [comment]
Closing as a fix has already been merged - @Linus789 please reopen if it still segfaults for you
